test/index.test.js

```diff
@@ -0,0 +1,159 @@
+
+      if (host.browser) {
+        // This assertion looks funky but because we're making a cross-origin request here we aren't going to have
+        // cookies present here even despite us sending `credentials: include`. We'll only be able to detect cookies
+        // here if we mock the server out, which we can't do in the browser.
+        //
+        // @todo we should try mocking this request instead to make sure that cookies are sent
+        expect(res.cookies).to.be.empty;
+      } else {
+        expect(res.cookies).to.deep.equal({
+          bar: 'baz',
+          foo: 'bar',
+        });
+      }
+    });
+
+    it('should support `application/x-www-form-urlencoded` requests with auth', async function () {
+      const res = await fetchHar(urlEncodedWithAuthHAR).then(r => r.json());
+
+      expect(res.args).to.deep.equal({ a: '1', b: '2' });
+      expect(res.data).to.equal('');
+      expect(res.files).to.be.empty;
+      expect(res.form).to.deep.equal({ category: '{"id":6,"name":"name"}', id: '8', name: 'name' });
+      expect(res.headers.Authorization).to.equal('Bearer api-key');
+      expect(res.headers['Content-Length']).to.equal('68');
+      expect(res.headers['Content-Type']).to.equal('application/x-www-form-urlencoded');
+      expect(res.json).to.be.null;
+      expect(res.url).to.equal('https://httpbin.org/post?a=1&b=2');
+    });
+
+    it('should support requests that cover the entire HAR spec', async function () {
+      const res = await fetchHar(harExamples.full).then(r => r.json());
+
+      expect(res.args).to.deep.equal({ baz: 'abc', foo: 'baz', key: 'value?foo=bar' });
+      expect(res.data).to.equal('');
+      expect(res.files).to.be.empty;
+      expect(res.form).to.deep.equal({ foo: 'bar' });
+      expect(res.headers['Content-Length']).to.equal('7');
+      expect(res.headers['Content-Type']).to.equal('application/x-www-form-urlencoded');
+
+      // @todo we should mock this request instead
+      if (host.node) {
+        expect(res.headers.Cookie).to.equal('foo=bar; bar=baz');
+      }
+
+      expect(res.json).to.be.null;
+      expect(res.url).to.equal('https://httpbin.org/post?key=value%3Ffoo=bar&foo=baz&baz=abc');
+    });
+
+    describe('multipart/form-data', function () {
+      it('should throw an error if `fileName` is present without `value` or a mapping', function () {
+        expect(() => {
+          fetchHar(harExamples['multipart-file']);
+        }).to.throw(/doesn't have access to the filesystem/);
+      });
+
+      describe('`files` option', function () {
+        it('should throw on an unsupported type', function () {
+          expect(() => {
+            fetchHar(harExamples['multipart-data-dataurl'], {
+              files: {
+                'owlbert.png': new Blob([owlbertDataURL], { type: 'image/png' }),
+              },
+            });
+          }).to.throw('An unknown object has been supplied into the `files` config for use.');
```

Comment:
I see a lot of `x-www-form-urlencoded` tests, and a single `form-data` test for unsupported types, do we have successful `form-data` tests somewhere I overlooked? If those are all in the quirks test file, that's not where I would expect them if I was trying to find a success test.

Reply:
We don't support `form-data` anymore, and if you use that with this library we'll throw an exception telling you to use `formdata-node` or `formdata-polyfill`.

Reply:
whoops, sorry I meant the `multipart/form-data` content type. I don't see a "this content type works fine" test, but I might have overlooked it.

Reply:
I'm adding tests for `postData.text` if `postData.params` isn't present right now, but for the "this content type works fine" that's part of these requests where we're looking at the responses. If `args` or `data` or `form` or `files` don't have what we're looking for, then the request's content type and payload wasn't sent right.

Reply:
For reference for httpbin these map to:

* args: query params
* data: raw body payloads
* form: multipart or `application/x-www-form-urlencoded`
* files: multpart with files